sympy/sets/fancysets.py

```diff
@@ -497,30 +497,38 @@ def normalize_theta_set(theta):
 
     if theta.is_Interval:
         # one complete circle
-        if Abs(theta.args[0] - theta.args[1]) >= 2*S.Pi:
+        intrvl_length = theta.measure
+        if intrvl_length >= 2*S.Pi:
+            if intrvl_length == 2*S.Pi and theta.left_open and theta.right_open:
+                k = coeff(theta.start)
+                return Union(Interval(0, k*S.Pi, False, True),
+                        Interval(k*S.Pi, 2*S.Pi, True, True))
             return Interval(0, 2*S.Pi, False, True)
 
         new_theta = []
-        for val in [theta.args[0], theta.args[1]]:
-            k = coeff(val)
+        for val in [(theta.start, theta.left_open), (theta.end, theta.right_open)]:
+            k = coeff(val[0])
             if (not k) and (k != S.Zero):
                 raise NotImplementedError('Normalizing theta without pi as'
                                           'coefficient, is not Implemented.')
             elif k == S.Zero:
-                if val == S.Zero:
-                    new_theta.append(S.Zero)
+                if val[0] == S.Zero:
+                    new_theta.append((S.Zero, val[1]))
                 else:
                     # when theta is n*pi
-                    new_theta.append(2*S.Pi)
+                    new_theta.append((2*S.Pi, val[1]))
```

Comment:
This seems to give a wrong value for an interval like ``[2*pi, 3*pi]``. Perhaps we should rather treat ``start`` and ``end`` separately.